src/main/java/com/hillel/items_exchange/service/impl/LocationServiceImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.hillel.items_exchange.service.impl;
+
+import com.hillel.items_exchange.dao.LocationRepository;
+import com.hillel.items_exchange.dto.LocationDto;
+import com.hillel.items_exchange.model.Location;
+import com.hillel.items_exchange.service.LocationService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static com.hillel.items_exchange.mapper.UtilMapper.convertAllTo;
+import static com.hillel.items_exchange.mapper.UtilMapper.convertTo;
+import static com.hillel.items_exchange.util.MessageSourceUtil.getExceptionMessageSource;
+
+@Service
+@RequiredArgsConstructor
+public class LocationServiceImpl implements LocationService {
+    private final LocationRepository locationRepository;
+
+    @Override
+    public List<LocationDto> findAll() {
+        return new ArrayList<>(convertAllTo(locationRepository.findAll(), LocationDto.class, ArrayList::new));
+    }
+
+    @Override
+    public Optional<Location> findById(long id) {
+        return locationRepository.findById(id);
+    }
+
+    @Override
+    public LocationDto getById(long id) throws ClassNotFoundException {
+        return findById(id)
+                .map(location -> convertTo(location, LocationDto.class))
+                .orElseThrow(() -> new ClassNotFoundException(getExceptionMessageSource("exception.illegal.id")));
```

Comment:
Why don't you return Optional.empty for such case instead of throwing the exception?